lib/router/Router.svelte

```diff
@@ -0,0 +1,213 @@
+<script>
+    import {onDestroy} from 'svelte'
```

Comment:
You probably have a linter installed that changed the formatting of this file entirely :( Because of that, it's pretty much impossible for us to review this...

Before this PR, could you open a separate one where you just run the linter (without any code changes)? Then we can at least see what's different.
(Feel free to include other changes such as the .gitignore ones if relevant)

Reply:
I do use prettier, but most of the noise is because of a code split of the router in dedicated classes (to better understand the code structure) and the switch to vite instead of rollup (I had issues making it work with Svelte 5).

I've restructured the branch to isolate the commit with the pure [svelte-5 migration here ](https://github.com/ItalyPaleAle/svelte-spa-router/pull/326/commits/774d406ec929793161a9c31d58cc1dd5f3cd1e8f) if that helps.

However, I've now opted to develop a custom router that is more suited to my needs, so I'm unsure if I'll be able to contribute much more to this PR.

Feel free to close it if you don't think it'll be helpful in the future :)